### PR TITLE
update precompilation and .gitignore

### LINKED ISSUE #2792: Clean up precompile statements
Before 1.2 release precompile statements need to be re-generated and tests should be cleaned up not to throw warnings under Julia 1.8.

I will do it once https://github.com/JuliaData/DataFrames.jl/pull/2791 is merged.

### PR DESCRIPTION
**Pull request by bkamins**
Fixes #2792

Only technical changes. After this is merged I will make a 1.2 release.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,8 +1,5 @@
-*.jl.cov
-*.jl.mem
```
Comment (nalimilan) [inline .gitignore:2]: Aren't these still useful?

Comment (bkamins) [reply .gitignore:2]: OK - re-introducing with an explanation.

**Comment by bkamins** [comment]
@nalimilan - I have re-added the requested files

**Review by nalimilan — APPROVED** [review]

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**